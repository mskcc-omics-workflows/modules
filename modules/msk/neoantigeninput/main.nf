process NEOANTIGENINPUT {
    tag "$meta.id"
    label 'process_single'
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://mskcc/neoantigeninputs:1.0.4':
        'docker.io/mskcc/neoantigeninputs:1.0.4' }"

    input:
    tuple val(meta),  path(inputMaf),      path(hlaFile)
    tuple val(meta2), path(phyloWGSsumm),  path(phyloWGSmut),   path(phyloWGSfolder)
    tuple val(meta3), path(mutNetMHCpan),  path(wtNetMHCpan)

    output:
    tuple val(meta), path("*_.json"),                                                  emit: json
    tuple val(meta), path("*.MUT.tsv"), path("*.WT.tsv"),                              emit: netMHCpanreformatted
    path "versions.yml",                                                               emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def id = task.ext.prefix ?: "${meta.id}"
    def patientid = task.ext.cohort ?: "${meta.id}_patient"
    def cohort = task.ext.cohort ?: "${meta.id}_cohort"

    """
        tree_folder_name=\$(basename -s .zip "${phyloWGSfolder}")
        mkdir \$tree_folder_name
        unzip ${phyloWGSfolder} -d \$tree_folder_name

        gzip -d -c ${phyloWGSsumm} > ${id}.summ.json
        gzip -d -c ${phyloWGSmut} > ${id}.mut.json



        python3 /usr/bin/eval_phyloWGS.py --maf_file ${inputMaf} \
        --summary_file ${id}.summ.json \
        --mutation_file ${id}.mut.json \
        --tree_directory \$tree_folder_name/ \
        --id ${id} --patient_id ${patientid} \
        --cohort ${cohort} --HLA_genes ${hlaFile} \
        --netMHCpan_MUT_input ${mutNetMHCpan} \
        --netMHCpan_WT_input ${wtNetMHCpan}
        ${args}


        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            neoantigeninput: \$(echo \$(python3 /usr/bin/eval_phyloWGS.py -v))
        END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def id = task.ext.prefix ?: "${meta.id}"
    def patientid =task.ext.cohort ?: "${meta.id}_patient"
    def cohort =task.ext.cohort ?: "${meta.id}_cohort"
    """

        touch ${patientid}_${id}_.json
        touch ${patientid}.MUT.tsv
        touch ${patientid}.WT.tsv
        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            neoantigeninput: \$(echo \$(python3 /usr/bin/eval_phyloWGS.py -v))
        END_VERSIONS
    """
}
