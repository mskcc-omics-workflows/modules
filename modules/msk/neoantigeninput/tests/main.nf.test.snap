{
    "neoantigeninput - json,tsv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_patient_test_.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_patient.MUT.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test_patient.WT.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,1d342cf8b0f1e381e7b0cba890948d23"
                ],
                "json": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_patient_test_.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "netMHCpanreformatted": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_patient.MUT.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test_patient.WT.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1d342cf8b0f1e381e7b0cba890948d23"
                ]
            }
        ],
        "timestamp": "2024-04-10T14:36:29.1748"
    },
    "neoantigeninput - json,tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_patient_test_.json:md5,4c08046a6f5826a3d51c221c6eef6348"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "netmHCpanoutput.MUT.tsv:md5,df56a2a876a57406b02f8664c28296ec",
                        "netmHCpanoutput.WT.tsv:md5,b1171286368f359f8bf854a7048e3e4c"
                    ]
                ],
                "2": [
                    "versions.yml:md5,1d342cf8b0f1e381e7b0cba890948d23"
                ],
                "json": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_patient_test_.json:md5,4c08046a6f5826a3d51c221c6eef6348"
                    ]
                ],
                "netMHCpanreformatted": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "netmHCpanoutput.MUT.tsv:md5,df56a2a876a57406b02f8664c28296ec",
                        "netmHCpanoutput.WT.tsv:md5,b1171286368f359f8bf854a7048e3e4c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1d342cf8b0f1e381e7b0cba890948d23"
                ]
            }
        ],
        "timestamp": "2024-04-10T14:36:22.508835"
    }
}